x-pack/plugins/ai_infra/product_doc_base/server/tasks/utils.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+import type { TaskManagerStartContract } from '@kbn/task-manager-plugin/server';
+
+export const getTaskStatus = async ({
+  taskManager,
+  taskId,
+}: {
+  taskManager: TaskManagerStartContract;
+  taskId: string;
+}) => {
+  try {
+    const taskInstance = await taskManager.get(taskId);
+    return taskInstance.status;
+  } catch (e) {
+    // not found means the task was completed and the entry removed
+    if (SavedObjectsErrorHelpers.isNotFoundError(e)) {
+      return 'not_scheduled';
+    }
+    throw e;
+  }
+};
+
+export const isTaskCurrentlyRunningError = (err: Error): boolean => {
+  return err.message?.includes('currently running');
```

Comment:
🙈 I'm surprised the task manager doesn't provide more robust utilities for these types of checks. Same for the `isNotFoundError` check above.

Reply:
Yeah I was too, and yet after checking with both Brandon and Mike, that kind of utilities are not exposed from TM atm